addon/-private/system/model/internal-model.js

```diff
@@ -882,20 +876,50 @@ export default class InternalModel {
   }
 
   /*
-    @method clearRelationships
+    This method should only be called by records in the `isNew()` state.
+    It will completely reset all relationships to an empty state and remove
+    the record from any associated inverses as well.
+
+    @method _resetAllRelationshipsToEmpty
     @private
   */
-  clearRelationships() {
+  _resetAllRelationshipsToEmpty() {
     this.eachRelationship((name, relationship) => {
       if (this._relationships.has(name)) {
         let rel = this._relationships.get(name);
+
         rel.clear();
         rel.removeInverseRelationships();
       }
     });
     Object.keys(this._implicitRelationships).forEach((key) => {
-      this._implicitRelationships[key].clear();
-      this._implicitRelationships[key].removeInverseRelationships();
+      let rel = this._implicitRelationships[key];
+
+      rel.clear();
+      rel.removeInverseRelationships();
+    });
+  }
+
+  /*
+    This method should only be called once the record has been deleted
+    and that deletion has been persisted. It will remove this record from
+    any associated relationships.
+
+    @method unloadDeletedRecordFromRelationships
+    @private
+   */
+  unloadDeletedRecordFromRelationships() {
```

Comment:
How about: `removeFromInverseRelationships`, example:

```js
internalModel.removeFromInverseRelationships();
```

The method would now describes concisely *what* is does not *when* to use it. The *when* and *why* should be provided via docs, or derived contextually from where it is used.